### PR TITLE
feat: Delete triggers

### PR DESCRIPTION
**Pull request by y-lohse**
Adds the ability to delete a `io.cozy.triggers`.

### CODE REVIEW

**Comment by y-lohse** [comment]
[(𝐒) [Connecteur Toutatice] 🎓- OIDC - logout](https://trello.com/c/zXear0YU/1959-%F0%9D%90%92-connecteur-toutatice-%F0%9F%8E%93-oidc-logout)

File: packages/cozy-stack-client/src/TriggerCollection.js [hunk 1]
```diff
@@ -69,8 +69,21 @@ class TriggerCollection {
     }
   }
 
-  async destroy() {
-    throw new Error('destroy() method is not available for triggers')
+  /**
+   * async destroy - Deletes a trigger
```
Comment (ptbrowne) [inline packages/cozy-stack-client/src/TriggerCollection.js]: I think it is unnecessary to repeat "async destroy".

File: packages/cozy-stack-client/src/TriggerCollection.js [hunk 2]
```diff
@@ -69,8 +69,21 @@ class TriggerCollection {
...
+   * @see https://docs.cozy.io/en/cozy-stack/jobs/#delete-jobstriggerstrigger-id
```
Comment (ptbrowne) [inline packages/cozy-stack-client/src/TriggerCollection.js]: cool !

File: packages/cozy-stack-client/src/TriggerCollection.spec.js [hunk 3]
```diff
@@ -97,4 +97,20 @@ describe('TriggerCollection', () => {
...
+      })
+      expect(stackClient.fetchJSON).toHaveBeenLastCalledWith(
+        'DELETE',
+        '/jobs/triggers/123'
+      )
+    })
+
+    it('should throw if the document has no _id', async () => {
+      await expect(collection.destroy({ _rev: 'yolo' })).rejects.toThrowError()
+    })
```
Comment (ptbrowne) [inline packages/cozy-stack-client/src/TriggerCollection.spec.js]: Can you add a test, it should add _deleted to the returned document ?

File: packages/cozy-stack-client/src/TriggerCollection.js [hunk 4]
```diff
@@ -69,8 +69,21 @@ class TriggerCollection {
...
+   * @param  {object} document The trigger to delete — must have an _id field
+   * @returns {object} The deleted document
+   */
+  async destroy(document) {
+    const { _id } = document
+    if (!_id) {
+      throw new Error('TriggerCollection.destroy needs a document with an _id')
+    }
+    await this.stackClient.fetchJSON('DELETE', uri`/jobs/triggers/${_id}`)
```
Comment (ptbrowne) [inline packages/cozy-stack-client/src/TriggerCollection.js]: I do think `uri` is needed here.

**Comment by ptbrowne** [comment]
LGTM

File: packages/cozy-stack-client/src/TriggerCollection.js (hunk 4 above)
Comment (y-lohse) [reply packages/cozy-stack-client/src/TriggerCollection.js]: Yep, we need to escape `_id` !

**Pull request merged**